CMakeLists.txt

```diff
@@ -283,43 +315,175 @@ if(UNIX AND NOT APPLE)
   set(MAYBE_RT_LIBRARY -lrt)
 endif()
 
-
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/cmake")
-set(OSMIUM_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/include")
-find_package(Osmium REQUIRED COMPONENTS io)
-add_dependency_includes(${OSMIUM_INCLUDE_DIR})
-
 # Disallow deprecated protozero APIs
 add_definitions(-DPROTOZERO_STRICT_API)
 
-find_package(Boost 1.49.0 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
+find_package(Threads REQUIRED)
+
+# if mason is enabled no find_package calls are made
+# to ensure that we are only compiling and linking against
+# fully portable mason packages
+if(ENABLE_MASON)
+  message(STATUS "Installing dependencies via mason")
+
+  set(Boost_USE_STATIC_LIBS ON)
+  mason_use(boost VERSION ${MASON_BOOST_VERSION} HEADER_ONLY)
+  add_dependency_includes(${MASON_PACKAGE_boost_INCLUDE_DIRS})
+  mason_use(boost_libfilesystem VERSION ${MASON_BOOST_VERSION})
+  set(Boost_FILESYSTEM_LIBRARY ${MASON_PACKAGE_boost_libfilesystem_STATIC_LIBS})
+  mason_use(boost_libiostreams VERSION ${MASON_BOOST_VERSION})
+  set(Boost_IOSTREAMS_LIBRARY ${MASON_PACKAGE_boost_libiostreams_STATIC_LIBS})
+  mason_use(boost_libprogram_options VERSION ${MASON_BOOST_VERSION})
+  set(Boost_PROGRAM_OPTIONS_LIBRARY ${MASON_PACKAGE_boost_libprogram_options_STATIC_LIBS})
+  mason_use(boost_libregex VERSION ${MASON_BOOST_VERSION})
+  set(Boost_REGEX_LIBRARY ${MASON_PACKAGE_boost_libregex_STATIC_LIBS})
+  mason_use(boost_libtest VERSION ${MASON_BOOST_VERSION})
+  set(Boost_UNIT_TEST_FRAMEWORK_LIBRARY ${MASON_PACKAGE_boost_libtest_STATIC_LIBS})
+  mason_use(boost_libdate_time VERSION ${MASON_BOOST_VERSION})
+  set(Boost_DATE_TIME_LIBRARY ${MASON_PACKAGE_boost_libdate_time_STATIC_LIBS})
+  mason_use(boost_libthread VERSION ${MASON_BOOST_VERSION})
+  set(Boost_THREAD_LIBRARY ${MASON_PACKAGE_boost_libthread_STATIC_LIBS})
+  mason_use(boost_libsystem VERSION ${MASON_BOOST_VERSION})
+  set(Boost_SYSTEM_LIBRARY ${MASON_PACKAGE_boost_libsystem_STATIC_LIBS})
+
+  mason_use(stxxl VERSION ${MASON_STXXL_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_stxxl_INCLUDE_DIRS})
+  set(STXXL_LIBRARY ${MASON_PACKAGE_stxxl_STATIC_LIBS})
+
+  mason_use(expat VERSION ${MASON_EXPAT_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_expat_INCLUDE_DIRS})
+  set(EXPAT_LIBRARIES ${MASON_PACKAGE_expat_STATIC_LIBS})
+
+  mason_use(lua VERSION ${MASON_LUA_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_lua_INCLUDE_DIRS})
+  set(USED_LUA_LIBRARIES ${MASON_PACKAGE_lua_STATIC_LIBS})
+
+  mason_use(luabind_lua524 VERSION ${MASON_LUABIND_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_luabind_lua524_INCLUDE_DIRS})
+  set(LUABIND_LIBRARY ${MASON_PACKAGE_luabind_lua524_STATIC_LIBS})
+
+  mason_use(bzip2 VERSION ${MASON_BZIP2_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_bzip2_INCLUDE_DIRS})
+  set(BZIP2_LIBRARIES ${MASON_PACKAGE_bzip2_STATIC_LIBS})
+
+  # todo - system packages do not provide a remote tarball
```

Comment:
@TheMarex https://github.com/mapbox/mason/issues/273. Per chat with @kkaefer the recommended approach is to just do `-lz` instead of using a "system" package from mason. But actually I think we should do `find_package(ZLIB` just like we are doing for non mason installs. I'll fix this up shortly.